File: EduLink.Domain/Strategies/TarjetaPagoStrategy.cs
using EduLink.Domain.Interfaces;
using System;
using System.Threading.Tasks;

namespace EduLink.Domain.Strategies;

public class TarjetaPagoStrategy : IPagoStrategy
{
    public string Nombre => "Tarjeta";
    public Task<bool> ProcesarAsync(decimal monto, Guid clienteId)
    {
        Console.WriteLine($"Procesando pago con tarjeta virtual de PeiGO por un monto de ${monto:F2}...");
        return Task.FromResult(true); // Hecho
    }
}

File: app/domain/strategies/tarjeta_pago_strategy.rb
require_relative '../interfaces/i_pago_strategy'

module Strategies
  class TarjetaPagoStrategy
    include Interfaces::IPagoStrategy

    def nombre
      "Tarjeta"
    end

    def procesar(monto, cliente_id)
      puts "Procesando pago con tarjeta por un monto de #{monto}..."
      # En una app real, aquí iría la lógica para conectar con un banco
      true # Simula un pago exitoso
    end
  end
end
